app/database/queries/breeze_buddy/merchant_entities.py

```
@@ -0,0 +1,407 @@
+    """Delete merchant entity by merchant_id.
+
+    Returns True if deleted, False if not found.
+    Raises ValueError if users still reference this merchant_id.
+    Raises exception on database errors.
+    """
+    # First check if any users reference this merchant_id
+    user_count = await check_users_with_merchant_id(merchant_id)
+    if user_count > 0:
+        raise ValueError(
+            f"Cannot delete merchant entity '{merchant_id}': {user_count} user(s) still reference it"
+        )
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)